### PR TITLE
call cleanup method of utils_netperf.Netperf explicitly

### PR DESCRIPTION
**Pull request by lolyu**
As the suject, aims to replace previous dependence on `__del__` to do
cleanup with explict call of cleanup function.

Signed-off-by: lolyu <lolyu@redhat.com>

### CODE REVIEW

**Comment by lolyu** [comment]
depends on https://github.com/avocado-framework/avocado-vt/pull/2084

File: qemu/tests/netperf_udp.py [hunk 1]
```diff
@@ -151,7 +151,7 @@ def run(test, params, env):
         netperf_server.start()
         # Run netperf with message size defined in range.
         msg = "Detail result of netperf test with different packet size.\n"
-        while(m_size <= end_size):
+        for m_size in range(start_size, end_size, step):
```
Comment (hereischen) [inline qemu/tests/netperf_udp.py]: Your change does not correct when `step = 1`, as `end_size` is excluded in `range`

Comment (lolyu) [reply qemu/tests/netperf_udp.py]: updated.

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Comment by lolyu** [comment]
```
2019-06-10 02:12:48,892 test             L1018 INFO | PASS 1-Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.netperf_udp
```

File: qemu/tests/netperf_udp.py [hunk 2]
```diff
@@ -169,9 +169,13 @@ def run(test, params, env):
...
             throughput.append(float(line_tokens[5]))
             msg += output
-            m_size += step
     finally:
+        error_context.context(
+            "Kill and cleanup netperf on both server and client.",
+            logging.info)
         netperf_server.stop()
+        netperf_server.cleanup()
+        netperf_client.cleanup()
```
Comment (luckyh) [inline qemu/tests/netperf_udp.py]: As errors may occur during the execution of `.cleanup`, we should consider whether let them terminate the test.

File: qemu/tests/ovs_qos.py [hunk 3]
```diff
@@ -244,6 +245,10 @@ def __get_vminfo():
                 clear_qos_setting(iface)
                 results.append([iface, throughout, rate, burst])
         report_test_results(results)
+        # cleanup netperf env
+        logging.debug("cleanup netperf env")
+        for ntpf, _ in itertools.chain(netperf_clients, netperf_servers):
+            ntpf.cleanup()
```
Comment (luckyh) [inline qemu/tests/ovs_qos.py]: This should go to the `finally` clause, right?

**Comment by lolyu** [comment]
``` 
(1/1) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.ovs_qos: PASS (241.41 s)
```
updated, please take a review @luckyh

File: qemu/tests/netperf_udp.py [hunk 4]
```diff
@@ -169,9 +169,15 @@ def run(test, params, env):
...
-            m_size += step
     finally:
+        logging.debug("Kill netperf server")
         netperf_server.stop()
+        try:
+            logging.debug("Cleanup env on both server and client")
+            netperf_server.cleanup()
+            netperf_client.cleanup()
+        except Exception as e:
+            logging.debug("Cleanup failed:\n%s\n", e)
```
Comment (luckyh) [inline qemu/tests/netperf_udp.py]: Should be `warning`?

**Comment by lolyu** [comment]
```
 (1/1) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.ovs_qos: PASS (240.40 s)
 (1/1) Host_RHEL.m7.u7.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.netperf_udp: PASS (544.56 s)
```

**Comment by pezhang** [comment]
Looks good to me.

Acked by: Pei Zhang pezhang@redhat.com

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**